lib/swaps/SwapClient.ts

```diff
@@ -142,43 +143,72 @@ abstract class SwapClient extends EventEmitter {
     });
   }
 
-  protected setStatus = async (status: ClientStatus): Promise<void> => {
+  public init = async () => {
+    // up front checks before initializing client
+    if (!this.isNotInitialized() && !this.isMisconfigured()) {
+      // we only initialize from NotInitialized or Misconfigured status
+      this.logger.warn(`can not init in ${this.status} status`);
+      return;
+    }
+    if (this.disable) {
+      this.setStatus(ClientStatus.Disabled);
+      return;
+    }
+
+    // client specific initialization
+    await this.initSpecific();
+
+    // final steps to complete initialization
+    this.setStatus(ClientStatus.Initialized);
+    this.setTimers();
+    this.emit('initialized');
+    await this.verifyConnectionWithTimeout();
+  }
+
+  protected abstract async initSpecific(): Promise<void>;
+
+  protected setConnected = async (newPubKey?: string, newUris?: string[]) => {
```

Comment:
`newPubKey` seems to always be present when calling `setConnected`. Should we also rename it to `newIdentifier`?

Reply:
Yes, it should be `newIdentifier`, but it won't always have a value. When the identifier doesn't change, it will be undefined. That logic hasn't changed in this PR.

Reply:
You're right. That logic didn't change :). Sorry.